contracts/0.8.9/StakingRouter.sol

```diff
@@ -0,0 +1,1109 @@
+    ///     node operator id
+    /// @param _stakingModuleId Id of the staking module
+    /// @param _nodeOperatorId Id of the node operator
+    /// @param _refundedValidatorsCount New number of refunded validators of the node operator
+    function updateRefundedValidatorsCount(
+        uint256 _stakingModuleId,
+        uint256 _nodeOperatorId,
+        uint256 _refundedValidatorsCount
+    ) external validStakingModuleId(_stakingModuleId) onlyRole(STAKING_MODULE_MANAGE_ROLE) {
+        address moduleAddr = _getStakingModuleById(_stakingModuleId).stakingModuleAddress;
+        IStakingModule(moduleAddr)
+            .updateRefundedValidatorsCount(_nodeOperatorId, _refundedValidatorsCount);
+    }
+
+    function reportRewardsMinted(uint256[] calldata _stakingModuleIds, uint256[] calldata _totalShares)
+        external
+        onlyRole(REPORT_REWARDS_MINTED_ROLE)
+    {
+        for (uint256 i = 0; i < _stakingModuleIds.length; ) {
+            address moduleAddr = _getStakingModuleById(_stakingModuleIds[i]).stakingModuleAddress;
+            IStakingModule(moduleAddr).handleRewardsMinted(_totalShares[i]);
+            unchecked { ++i; }
+        }
+    }
+
+    function updateExitedValidatorsCountByStakingModule(
+        uint256[] calldata _stakingModuleIds,
+        uint256[] calldata _exitedValidatorsCounts
+    )
+        external
+        onlyRole(REPORT_EXITED_VALIDATORS_ROLE)
+    {
+        for (uint256 i = 0; i < _stakingModuleIds.length; ) {
+            StakingModule storage stakingModule = _getStakingModuleById(_stakingModuleIds[i]);
+            uint256 prevReportedExitedValidatorsCount = stakingModule.exitedValidatorsCount;
+            if (_exitedValidatorsCounts[i] < prevReportedExitedValidatorsCount) {
+                revert ErrorExitedValidatorsCountCannotDecrease();
+            }
+
+            (
+                uint256 totalExitedValidatorsCount,
+                /* uint256 totalDepositedValidators */,
+                /* uint256 depositableValidatorsCount */
+            ) = IStakingModule(stakingModule.stakingModuleAddress).getStakingModuleSummary();
+
+            if (totalExitedValidatorsCount < prevReportedExitedValidatorsCount) {
+                // not all of the exited validators were async reported to the module
+                emit StakingModuleExitedValidatorsIncompleteReporting(
+                    stakingModule.id,
+                    prevReportedExitedValidatorsCount - totalExitedValidatorsCount
+                );
+            }
+            stakingModule.exitedValidatorsCount = _exitedValidatorsCounts[i];
+            unchecked { ++i; }
+        }
+    }
+
+    function reportStakingModuleExitedValidatorsCountByNodeOperator(
+        uint256 _stakingModuleId,
+        bytes calldata _nodeOperatorIds,
+        bytes calldata _exitedValidatorsCounts
+    )
+        external
+        onlyRole(REPORT_EXITED_VALIDATORS_ROLE)
```

Comment:
maybe rename it to `REPORT_EXITED_AND_STUCK_VALIDATORS_ROLE` or have a separate one